fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/AbstractService.java

```diff
@@ -286,7 +286,7 @@ protected void lockArchivalGroupResource(final Transaction tx,
                                              final FedoraId fedoraId) {
         final var headers = pSession.getHeaders(fedoraId, null);
         if (headers.getArchivalGroupId() != null) {
-            tx.lockResource(headers.getArchivalGroupId());
+            tx.lockResourceAndGhostNodes(headers.getArchivalGroupId());
```

Comment:
Do you think we need to look for ghosts here? Let's say that we have a ghost node `a` that contains the AG `a/b` and we're adding `a/b/c`. I think that locking `a/b` and `a/b/c` is sufficient because the ghost node is not actually being created here.

Reply:
So because this resource is part of an AG, then that AG must already be created or locked?

Reply:
Yes, I think that we do not need to look for ghosts in the ancestry of the parent of the target resource. I think we only care about ghosts that are between the parent and the target. But, maybe I'm wrong.